File: app/src/main/java/pt/ipbeja/pi/piproject/identificationInsect/MoreInfoPopupDialog.java
package pt.ipbeja.pi.piproject.identificationInsect;

import android.app.Activity;
import android.app.Dialog;
import android.content.Intent;
import android.content.res.AssetManager;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import java.io.IOException;
import java.io.InputStream;

import pt.ipbeja.pi.piproject.R;
import pt.ipbeja.pi.piproject.idkey.KeyOption;

/**
 * Created by Vicenzo on 01-02-18.
 */

public class MoreInfoPopupDialog extends Dialog {
    public Activity activity;
    public Dialog dialog;
    public Button close;
    public KeyOption option;

    public MoreInfoPopupDialog(Activity a, KeyOption option) {
        super(a);
        //Log.e("$$$$$$$$$$$$->", "Info");
        this.activity = a;
        this.option = option;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.dialog_moreinfopopup);

        close = (Button) findViewById(R.id.closeBtn);

        close.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v)
            {
                dismiss();
            }
        });


        ImageView image = (ImageView) findViewById(R.id.imgView);

        AssetManager assMan = activity.getAssets();
        String imagePath = option.getImageLocation();
        try {
            InputStream ims = assMan.open(imagePath);
            Bitmap bitmap = BitmapFactory.decodeStream(ims);
            image.setImageBitmap(bitmap);
            ims.close();
        }
        catch(IOException e)
        {
            e.printStackTrace();
        }

        TextView descText = (TextView) findViewById(R.id.descLbl);
        String description = option.getDescription();
        descText.setText(description);

        image.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MoreInfoPopupDialog.this.activity, ImageZoom.class);
                intent.putExtra("optionImage", MoreInfoPopupDialog.this.option.getImageLocation());
                MoreInfoPopupDialog.this.activity.startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/pt/ipbeja/pi/piproject/identificationInsect/MoreInfoPopupDialog.kt
package pt.ipbeja.pi.piproject.identificationInsect

import android.app.Activity
import android.app.Dialog
import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.View
import android.view.Window
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import pt.ipbeja.pi.piproject.R
import pt.ipbeja.pi.piproject.idkey.KeyOption
import java.io.IOException

/**
 * Created by Vicenzo on 01-02-18.
 */
class MoreInfoPopupDialog //Log.e("$$$$$$$$$$$$->", "Info");
    (var activity: Activity, var option: KeyOption) : Dialog(activity) {
    var dialog: Dialog? = null
    var close: Button? = null

    override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.dialog_moreinfopopup)

        close = findViewById<View>(R.id.closeBtn) as Button

        close!!.setOnClickListener { dismiss() }


        val image = findViewById<View>(R.id.imgView) as ImageView

        val assMan = activity.assets
        val imagePath = option.imageLocation
        try {
            val ims = assMan.open(imagePath)
            val bitmap = BitmapFactory.decodeStream(ims)
            image.setImageBitmap(bitmap)
            ims.close()
        } catch (e: IOException) {
            e.printStackTrace()
        }

        val descText = findViewById<View>(R.id.descLbl) as TextView
        val description = option.description
        descText.text = description

        image.setOnClickListener {
            val intent = Intent(
                this@MoreInfoPopupDialog.activity,
                ImageZoom::class.java
            )
            intent.putExtra(
                "optionImage",
                option.imageLocation
            )
            activity.startActivity(intent)
        }
    }
}